From ed520a303e02e116cdaed49d9960f4009e613caf Mon Sep 17 00:00:00 2001
From: George T Kramer <george.t.kramer@intel.com>
Date: Wed, 16 May 2018 03:16:05 +0000
Subject: [PATCH] Use fPIC as recommended by ld

---
 ffi/build.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ffi/build.py b/ffi/build.py
index 9efb189..bc4f57f 100755
--- a/ffi/build.py
+++ b/ffi/build.py
@@ -127,7 +127,7 @@ def main_posix(kind, library_ext):
     cxxflags = run_llvm_config(llvm_config, ["--cxxflags"])
     # on OSX cxxflags has null bytes at the end of the string, remove them
     cxxflags = cxxflags.replace('\0', '')
-    cxxflags = cxxflags.split() + ['-fno-rtti', '-g']
+    cxxflags = cxxflags.split() + ['-fno-rtti', '-g', '-fPIC']
 
     # look for SVML
     include_dir = run_llvm_config(llvm_config, ['--includedir']).strip()
-- 
2.23.0

