From fab4a09362cd746e85e9fc7bc6f411ebd4a92e28 Mon Sep 17 00:00:00 2001
From: George T Kramer <george.t.kramer@intel.com>
Date: Wed, 16 May 2018 03:16:05 +0000
Subject: [PATCH] Use fPIC as recommended by ld

---
 ffi/build.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ffi/build.py b/ffi/build.py
index e6226d2..9680949 100755
--- a/ffi/build.py
+++ b/ffi/build.py
@@ -112,7 +112,7 @@ def main_posix(kind, library_ext):
     cxxflags = run_llvm_config(llvm_config, ["--cxxflags"])
     # on OSX cxxflags has null bytes at the end of the string, remove them
     cxxflags = cxxflags.replace('\0', '')
-    cxxflags = cxxflags.split() + ['-fno-rtti', '-g']
+    cxxflags = cxxflags.split() + ['-fno-rtti', '-g', '-fPIC']
     os.environ['LLVM_CXXFLAGS'] = ' '.join(cxxflags)
 
     ldflags = run_llvm_config(llvm_config, ["--ldflags"])
-- 
2.17.0

